### PR TITLE
fix helm chart gateway proxy rendering

### LINKED ISSUE #4528: custom gateway proxies don't override values from the defaul gateway proxy
**Describe the bug**
When defining a custom gatewayProxy, the default values are not correctly overridden.

**To Reproduce**

The following helm values will not render any `Gateway` resource

```yaml
 gatewayProxies:
    gatewayProxy:
      disabled: true

    gatewayProxyPublic:
      disabled: false
```

**Expected behavior**

a `Gateway` resource named `gateway-proxy-public` should be created

**Additional context**
Add any other context about the problem here, e.g.
- Gloo Edge version 1.7.0-rc2
- Kubernetes version

### LINKED ISSUE #4529: Helm: malformed yaml is rendered when attempting to create multiple gateway proxies
**Describe the bug**
When defining multiple `gatewayProxy` in a helm values file, malformed yaml is rendered

**To Reproduce**
Define multiple gateway proxies

```yaml
 gatewayProxies:
    gatewayProxy:
      disabled: false

    gatewayProxyPublic:
      disabled: false
```

Helm renders the following yaml

```yaml
---apiVersion: gateway.solo.io/v1
kind: Gateway
```

**Expected behavior**

`apiVersion` should be on a new line

```yaml
---
apiVersion: gateway.solo.io/v1
kind: Gateway
```

**Additional context**
Add any other context about the problem here, e.g.
- Gloo Edge version 1.7.0-rc2
- Kubernetes version

### PR DESCRIPTION
**Pull request by jon-walton**
# Description

The helm chart currently renders the https gateway incorrectly resulting in an error when being applied

```yaml
---apiVersion: gateway.solo.io/v1
kind: Gateway
```

should be

```yaml
---
apiVersion: gateway.solo.io/v1
kind: Gateway
```

it also fixes an issue with custom gateway proxies not correctly overwriting values from the default proxy, such as

```yaml
 gatewayProxies:
    gatewayProxy:
      disabled: true

    gatewayProxyPublic:
      disabled: false
```

`gatewayProxyPublic` would still be disabled

# Context

fixes #4529
fixes #4528

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4528
resolves https://github.com/solo-io/gloo/issues/4529

### CODE REVIEW

**Comment by jon-walton** [comment]
sorry, i'm not sure what folder to put the changelog in. there's only a v1.7.0-rc1 which has already been released. do you want me to create an rc2?

**Comment by kdorosh** [comment]
@jon-walton yes, please create an rc2 folder. Thanks for the contribution!

File: install/helm/gloo/templates/8-gateway-proxy-pod-disruption-budget.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {{- if .Values.gateway.enabled }}
 {{- $gatewayProxy := .Values.gatewayProxies.gatewayProxy -}}
 {{- range $name, $gatewaySpec := .Values.gatewayProxies }}
-{{- $spec := deepCopy $gatewayProxy | merge $gatewaySpec -}}
+{{- $spec := deepCopy $gatewaySpec | mergeOverwrite (deepCopy $gatewayProxy) -}}
```
Comment (kdorosh) [inline install/helm/gloo/templates/8-gateway-proxy-pod-disruption-budget.yaml:4]: we should have a helm test for this fix, see https://github.com/solo-io/gloo/blob/526805af36e878f8c45eb1e274e93a1cc19d08dd/install/test/helm_test.go#L1313 for example helm tests

File: install/helm/gloo/templates/8-default-gateways.yaml [hunk 2]
```diff
@@ -43,7 +43,7 @@ spec:
   - {{ $name | kebabcase }}
 {{- end }}
 ---
-{{- if not $gatewaySettings.disableHttpsGateway -}}
+{{- if not $gatewaySettings.disableHttpsGateway }}
```
Comment (kdorosh) [inline install/helm/gloo/templates/8-default-gateways.yaml:46]: 🎉

File: install/helm/gloo/templates/8-gateway-proxy-pod-disruption-budget.yaml (hunk 1 above)
Comment (jon-walton) [reply install/helm/gloo/templates/8-gateway-proxy-pod-disruption-budget.yaml:4]: Awesome, I can get you that tomorrow (I'm utc+8)

**Comment by jon-walton** [comment]
hi @kdorosh , apologies for the wait. I've added the unit tests (they fail without the helm chart fix)

**Comment by jon-walton** [comment]
for the changelog, do I just link to this pr? (there isn't an issue)

**Comment by EItanya** [comment]
> for the changelog, do I just link to this pr? (there isn't an issue)

Can you make an issue to reflect this?

Thanks

**Comment by jon-walton** [comment]
> 
> 
> > for the changelog, do I just link to this pr? (there isn't an issue)
> 
> Can you make an issue to reflect this?
> 
> Thanks

done

**Comment by kdorosh** [comment]
let's update the changelog to directory to v1.8.0-beta3 now that v1.8.0-beta2 has been released.

/test

**Comment by jon-walton** [comment]
@kdorosh done! 🥳

**Review by kdorosh — APPROVED** [review]

**Comment by kdorosh** [comment]
Thanks for the contribution!

**Comment by alex-berger** [comment]
Will this be back-ported to version 1.7.* (e.g. 1.7.1) as otherwise version 1.7.* is pretty useless for those who are affected by this.

**Comment by kdorosh** [comment]
@alex-berger thanks for reminder, see https://github.com/solo-io/gloo/pull/4552

**Pull request merged**